tests/unit/TechRecordsService.unitTest.ts

```diff
@@ -785,3 +878,142 @@ describe("updateEuVehicleCategory", () => {
+      expect(err.body).toEqual(ERRORS.STATUS_CODE_SHOULD_BE_PROVISIONAL);
+    });
+  });
+  });
+
+  context("a new Provisional tech record is created for a vehicle, that does not have a 'current' tech record and has an existing 'provisional'", () => {
+    it("should not allow to save the tech record", async () => {
+    const techRecord: any = cloneDeep(records[43]);
+    techRecord.techRecord[0].statusCode = STATUS.PROVISIONAL;
+    const payload: any = cloneDeep(records[43]);
+
+    const MockDAO = jest.fn().mockImplementation(() => {
+      return {
+        getBySearchTerm: () => {
+          return Promise.resolve({
+            Items: [cloneDeep(techRecord)],
+            Count: 1,
+            ScannedCount: 1
+          });
+        }
+      };
+    });
+    expect.assertions(2);
+    const techRecordsService = new TechRecordsService(new MockDAO());
+    techRecordsService.addNewProvisionalRecord(payload,msUserDetails).catch( (err) => {
+      expect(err.statusCode).toEqual(400);
+      expect(err.body).toEqual(ERRORS.CURRENT_OR_PROVISIONAL_RECORD_FOUND);
+    });
+  });
+  });
+});
+
+describe("archiveTechRecordStatus", () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+  const msUserDetails = {
+    msUser: "dorel",
+    msOid: "1234545"
+  };
+  context("A technical record status is updated to archived via PUT verb", () => {
+    it("should change the records status to archived and set audit details and create no other record", async () => {
+    const techRecord: any = cloneDeep(records[43]);
+    const expectedTechRecord = cloneDeep(techRecord);
+    expectedTechRecord.techRecord[0].statusCode = STATUS.ARCHIVED;
+
+    const MockDAO = jest.fn().mockImplementation(() => {
+      return {
+        updateSingle: () => {
+          return Promise.resolve({
+            Attributes: expectedTechRecord
+          });
+        },
+        getBySearchTerm: () => {
+          return Promise.resolve({
+            Items: [cloneDeep(records[43])],
+            Count: 1,
+            ScannedCount: 1
+          });
+        }
+      };
+    });
+    expect.assertions(2);
+    const techRecordsService = new TechRecordsService(new MockDAO());
+    const updatedTechRec: any = await techRecordsService.archiveTechRecordStatus(techRecord.systemNumber, techRecord, msUserDetails);
+    expect(updatedTechRec.techRecord[0].statusCode).toEqual(STATUS.ARCHIVED);
+    expect(updatedTechRec.techRecord.length).toEqual(1);
+  });
+  });
+
+  context("A technical record status is updated to archived via PUT verb", () => {
+    it("should change the records status to archived and set audit details and create no other record", async () => {
+    const techRecord: any = cloneDeep(records[43]);
+    techRecord.techRecord[0].statusCode = STATUS.ARCHIVED;
+    const MockDAO = jest.fn().mockImplementation(() => {
+      return {
+        getBySearchTerm: () => {
+          return Promise.resolve({
+            Items: [cloneDeep(techRecord)],
```

Comment:
Can we use techRecord here or any particular reason we are cloning the object again?

Reply:
removed the cloning call